lib/active_record_host_pool/connection_adapter_mixin.rb

```diff
@@ -1,10 +1,15 @@
 # frozen_string_literal: true
 
-case ActiveRecordHostPool.loaded_db_adapter
-when :mysql2
+begin
   require "active_record/connection_adapters/mysql2_adapter"
-when :trilogy
+rescue LoadError
```

Comment:
Are you sure this won’t be a `Gem::MissingSpecError`, raised from e.g. https://github.com/rails/rails/blob/26bc62a795a2627026af7574e350c2d29eda54ef/activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb#L6 ? We cannot really test this case, right?

Reply:
I think it should work although it isn't really testable without adding two separate sets of gemfiles again.

```
begin
  gem "does_not_exist", "~> 2.0"
rescue LoadError => e
  puts e.class
  puts e.class.ancestors.include?(LoadError)
end =>

Gem::LoadError
true
```